app/src/main/java/org/simple/clinic/contactpatient/ContactPatientBottomSheet.kt

```diff
@@ -130,6 +134,14 @@ class ContactPatientBottomSheet : BaseBottomSheet<
     context.injector<Injector>().inject(this)
   }
 
+  override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+    val dialog = super.onCreateDialog(savedInstanceState) as BottomSheetDialog
+
+    dialog.overrideCancellation(::backPressed)
+
+    return dialog
```

Comment:
We already are doing this in `BaseBottomSheet`. 🤔 

If you need access to the `Back Press` event, I would suggest making `onBackPressed` a protected method instead and overriding it (while enforcing a call to the super method.

Reply:
I thought about that. But I felt like this was much more explicit and we don't end up easily missing the super call in `onBackPressed`. Also when we call `backPressed` we are doing a back press-on activity, so it can cause problems down the line when someone doesn't know to make the super call for it? WDYT?

Reply:
Another reason is, it seemed much more in line with the `OnBackPressedDispatcher` implementation Android has.

Reply:
One reason we wanted to implement this in the base class is because we need the `Router` to know that this screen is going to be removed from the hierarchy. Is this not required in this case?

Reply:
It is required in certain scenarios. Currently, we are overriding the back click to handle the UI changes in this sheet, so in certain cases, we won't remove the sheet from the back stack but simply update the UI to go back to the previous UI mode. Once we separate the 3 UI modes into their own sheets/screens, we don't have to override this.